network/endpoint_windows.go

```diff
@@ -544,3 +621,103 @@ func (epInfo *EndpointInfo) GetEndpointInfoByIPImpl(ipAddresses []net.IPNet, net
+	if err != nil {
+		logger.Error("Failed to disable VF device", zap.Error(err))
+		errMsg := fmt.Sprintf("Failed to disable VF device due to error %s", err.Error())
+		return errors.Errorf(errMsg)
+	}
+
+	logger.Info("Successfully disabled", zap.String("VF device", instanceID))
+	return nil
+}
+
+// Dismount VF device
+func DisamountVFDevice(instanceID string, plc platform.ExecClient) error {
+	locationPath, err := GetLocationPath(instanceID, plc)
+	if err != nil {
+		return err
+	}
+
+	disamountVFDevice := fmt.Sprintf("Dismount-VMHostAssignableDevice -Force -LocationPath \"%s\" -confirm:$false", locationPath) //nolint
+	_, err = plc.ExecutePowershellCommand(disamountVFDevice)
+	if err != nil {
+		logger.Error("Failed to dismount VF device", zap.Error(err))
+		errMsg := fmt.Sprintf("Failed to disamount VF device due to error %s", err.Error())
+		return errors.Errorf(errMsg)
+	}
+
+	logger.Info("Successfully dismounted", zap.String("VF device", instanceID))
+	return nil
+}
+
+// Get LocationPath
+func GetLocationPath(instanceID string, plc platform.ExecClient) (string, error) {
+	getLocationPath := fmt.Sprintf("(Get-PnpDeviceProperty -KeyName DEVPKEY_Device_LocationPaths –InstanceId \"%s\").Data[0]", instanceID) //nolint
+	locationPath, err := plc.ExecutePowershellCommand(getLocationPath)
+	if err != nil {
+		logger.Error("Failed to get VF locationPath", zap.Error(err))
+		errMsg := fmt.Sprintf("Failed to get VF locationPath due to error %s", err.Error())
+		return "", errors.Errorf(errMsg)
+	}
+
+	logger.Info("Successfully got", zap.String("location path", locationPath))
+	return locationPath, nil
+}
+
+// Get PnP device state
+// return devpkeyDeviceIsPresent and devpkeyDeviceProblemCode
+func GetPnpDeviceState(instanceID string, plc platform.ExecClient) (string, string, error) { //nolint
+	getDeviceIsPresent := fmt.Sprintf("(Get-PnpDeviceProperty -InstanceId \"%s\" | Where-Object KeyName -eq DEVPKEY_Device_IsPresent).Data[0]", instanceID) //nolint
+	devpkeyDeviceIsPresent, err := plc.ExecutePowershellCommand(getDeviceIsPresent)
+	if err != nil {
+		logger.Error("Failed to get if PnP device present or not", zap.Error(err))
+		errMsg := fmt.Sprintf("Failed to get if PnP device present or not due to error %s", err.Error())
+		return "", "", errors.Errorf(errMsg)
+	}
+	logger.Info("Successfully got", zap.String("device state presence", devpkeyDeviceIsPresent))
+
+	// DEVPKEY_Device_ProblemCode is not there once device is disabled and dismounted, so need to check if DEVPKEY_Device_ProblemCode exists first
+	getDeviceProblemCodeExist := fmt.Sprintf("(Get-PnpDeviceProperty -InstanceId \"%s\" | Where-Object KeyName -eq DEVPKEY_Device_ProblemCode)", instanceID) //nolint
+	devpkeyDeviceProblemCodeExist, _ := plc.ExecutePowershellCommand(getDeviceProblemCodeExist)
+	// only return isPresent flag and empty string as problemCode
+	if devpkeyDeviceProblemCodeExist == "" {
+		return devpkeyDeviceIsPresent, "", nil
+	}
```

Comment:
whats the value of `devpkeyDeviceIsPresent` in this case if get device property not succeeded and problem code is empty?

Reply:
i think we should return error saying problem code unknown or something